internal/files/copy.go

```diff
@@ -5,24 +5,34 @@
 package files
 
 import (
+	"fmt"
 	"os"
 	"path/filepath"
+	"regexp"
 
 	"github.com/magefile/mage/sh"
 )
 
 // CopyAll method copies files from the source to the destination skipping empty directories.
 func CopyAll(sourcePath, destinationPath string) error {
-	return CopyWithSkipped(sourcePath, destinationPath, []string{})
+	return CopyWithSkipped(sourcePath, destinationPath, []string{}, []string{})
 }
 
 // CopyWithoutDev method copies files from the source to the destination, but skips _dev directories and empty folders.
 func CopyWithoutDev(sourcePath, destinationPath string) error {
-	return CopyWithSkipped(sourcePath, destinationPath, []string{"_dev", "build", ".git"})
+	return CopyWithSkipped(sourcePath, destinationPath, []string{"_dev", "build", ".git"}, []string{"\\.DS_Store", "\\..*\\.swp"})
 }
 
-// CopyWithSkipped method copies files from the source to the destination, but skips selected directories and empty folders.
-func CopyWithSkipped(sourcePath, destinationPath string, skippedDirs []string) error {
+// CopyWithSkipped method copies files from the source to the destination, but skips selected directories, empty folders and selected hidden files.
+func CopyWithSkipped(sourcePath, destinationPath string, skippedDirs, skippedFiles []string) error {
+	regexesFiles := []*regexp.Regexp{}
+	for _, regexFile := range skippedFiles {
+		r, err := regexp.Compile(regexFile)
+		if err != nil {
+			return fmt.Errorf("failed to compile regex %q: %w", r, err)
+		}
+		regexesFiles = append(regexesFiles, r)
+	}
```

Comment:
Pre-compile all regexes before calling to the `Walk()` function

Reply:
Updated the code to use glob patterns instead of regexes:
https://github.com/elastic/elastic-package/pull/2471/commits/b91048e4220ce27f58d02df3eb2c4228266aefd0

As everything should be file names glob, patterns should be enough.